src/scene/SceneControls.tsx

```diff
@@ -95,6 +97,10 @@ const SceneControls: React.FC<{
     return isTransitioningToPrevious || isTransitioningToNext;
   }, [isTransitioningToNext, isTransitioningToPrevious]);
 
+  useEffect(() => {
+    onTransition(isTransitioning);
+  }, [isTransitioning, onTransition]);
```

Comment:
Should we just lift this into `SceneManager` and pass it down? Feels like more of a scene manager responsibility anyways.